### PR TITLE
Extract json-api-client into separate package

### PR DESCRIPTION
**Pull request by rupurt**

### CODE REVIEW

File: tools/json-api-client/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@chainlink/json-api-client",
+  "private": true,
+  "version": "0.0.0",
+  "main": "./dist/src",
+  "types": "./dist/src",
+  "scripts": {
+    "build": "rm -rf dist && tsc",
+    "lint": "eslint --ext .js,.ts .",
+    "format": "prettier --write \"*.js\"",
```
Comment (HenryNguyen5) [inline tools/json-api-client/package.json]:
```suggestion
    "format": "prettier --write \"*.ts\"",
```

File: tools/json-api-client/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@chainlink/json-api-client",
+  "private": true,
+  "version": "0.0.0",
```
Comment (HenryNguyen5) [inline tools/json-api-client/package.json]:
```suggestion
  "version": "0.0.1",
```

File: tools/json-api-client/package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  "main": "./dist/src",
+  "types": "./dist/src",
```
Comment (HenryNguyen5) [inline tools/json-api-client/package.json]: I don't think you need this field if they're in the same place. Could be wrong though

File: tools/json-api-client/package.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  "scripts": {
+    "build": "rm -rf dist && tsc",
```
Comment (HenryNguyen5) [inline tools/json-api-client/package.json]: This isn't windows friendly. Should we change it?

```suggestion
    "build": "rimraf dist && tsc",
```
Would need to add the `rimraf` dependency

File: tools/json-api-client/src/formatRequestURI.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import url, { UrlObject } from 'url'
+
+export interface Options {
+  port?: string
+  hostname?: string
+}
+
+export default (path: string, query = {}, options: Options = {}) => {
```
Comment (HenryNguyen5) [inline tools/json-api-client/src/formatRequestURI.ts]: Can we convert this to a function declaration?

Comment (HenryNguyen5) [reply tools/json-api-client/src/formatRequestURI.ts]: Query should probably be of type `Record<string,string>`

File: tools/json-api-client/src/formatRequestURI.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  const formatOptions: UrlObject = {
+    pathname: path,
+    query: query,
```
Comment (HenryNguyen5) [inline tools/json-api-client/src/formatRequestURI.ts]:
```suggestion
    query,
```

File: tools/json-api-client/src/formatRequestURI.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import url, { UrlObject } from 'url'
+
+export interface Options {
+  port?: string
+  hostname?: string
+}
+
+export default (path: string, query = {}, options: Options = {}) => {
+  const formatOptions: UrlObject = {
+    pathname: path,
```
Comment (HenryNguyen5) [inline tools/json-api-client/src/formatRequestURI.ts]: Maybe rename this so we can have shorthand object properties?

File: tools/json-api-client/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
+{
+  "compilerOptions": {
+    "composite": true,
+    "declaration": true,
+    "declarationMap": true,
+    "strict": true,
+    "skipLibCheck": true,
```
Comment (HenryNguyen5) [inline tools/json-api-client/tsconfig.json]: Can we remove this?

File: tools/json-api-client/tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    "noEmitOnError": true,
+    "allowSyntheticDefaultImports": true,
+    "moduleResolution": "node",
+    "target": "es2017",
+    "module": "es2015",
+    "lib": ["es2018", "dom"],
```
Comment (HenryNguyen5) [inline tools/json-api-client/tsconfig.json]: Is this only going to be used client-side? If not, remove "dom"

File: tools/json-api-client/package.json (hunk 3 above)
Comment (rupurt) [reply tools/json-api-client/package.json]: This comes from https://github.com/smartcontractkit/chainlink/blob/develop/styleguide/package.json#L6

File: tools/json-api-client/package.json (hunk 4 above)
Comment (rupurt) [reply tools/json-api-client/package.json]: This comes from https://github.com/smartcontractkit/chainlink/blob/develop/styleguide/package.json#L10

File: tools/json-api-client/package.json (hunk 3 above)
Comment (HenryNguyen5) [reply tools/json-api-client/package.json]: Lets just keep it for consistency then

File: tools/json-api-client/package.json (hunk 4 above)
Comment (HenryNguyen5) [reply tools/json-api-client/package.json]: Lets just keep it for consistency then

File: tools/json-api-client/tsconfig.json (hunk 8 above)
Comment (rupurt) [reply tools/json-api-client/tsconfig.json]: Don't we need this for `global` & `json-api-normalizer`?

File: tools/json-api-client/tsconfig.json (hunk 9 above)
Comment (rupurt) [reply tools/json-api-client/tsconfig.json]: I would like to use it in CLI's if we could

File: tools/json-api-client/tsconfig.json (hunk 8 above)
Comment (HenryNguyen5) [reply tools/json-api-client/tsconfig.json]: We shouldn't no, this is a relic of having our yarn workspaces mashed up.

File: tools/json-api-client/tsconfig.json (hunk 9 above)
Comment (HenryNguyen5) [reply tools/json-api-client/tsconfig.json]: Yeah, lets remove this then so we dont make any DOM assumptions by accident

File: tools/json-api-client/package.json [hunk 10]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@chainlink/json-api-client",
+  "private": true,
+  "version": "0.0.1",
+  "main": "./dist/src",
+  "types": "./dist/src",
+  "scripts": {
+    "build": "rm -rf dist && tsc",
+    "lint": "eslint --ext .js,.ts .",
```
Comment (HenryNguyen5) [inline tools/json-api-client/package.json]: I think we can remove the `.js` extension here

**Pull request merged**